### PR TITLE
Special Fixes for Jenkins Worker

### PR DESCRIPTION
**Pull request by shadinaif**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by shadinaif** [comment]
@OmarEdraak @devalih 
Please review this PR as Jenkins Worker setup did not succeed without these changes.

We have one problem as far as I can tell:

Edx team has upgraded the ansible version. Jenkins Worker is failing because of the new 2.3.1 version. This PR downgraded the version again, but this will brake something outside the Jenkins Worker setup.

We can stick with the new version; but that will require many updates on the workers yml files. Everything checking for a variable definition must be reviewed and many will need to be refactored. It is a bug in 2.3 version. Upgrading to 2.4 will solve this problem but it will create other problems

File: playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml [hunk 1]
```diff
@@ -36,32 +36,6 @@
     chdir: "{{ jenkins_home }}"
   become_user: "{{ jenkins_user }}"
 
-- name: Add script to set install node packages
```
Comment (OmarEdraak) [inline playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml]: @shadinaif this is usually needed for running the front end tests. I don't think we should merge this pull request with this change.

File: playbooks/roles/jenkins_worker/tasks/test_platform_worker.yml [hunk 2]
```diff
@@ -34,8 +34,8 @@
       - "{{ testfile.stat.mtime }} - {{ edxvenv.stat.mtime }} < 3600"
 
 # Verify that postfix (mail server) is listening
-- wait_for: host={{ inventory_hostname }} port=25
-  delegate_to: localhost
+#- wait_for: host={{ inventory_hostname }} port=25
+#  delegate_to: localhost
```
Comment (OmarEdraak) [inline playbooks/roles/jenkins_worker/tasks/test_platform_worker.yml]: haha, classic error, yes we don't need it!

File: playbooks/roles/jenkins_worker/templates/jenkins_nodeenv.j2 [hunk 3]
```diff
@@ -1,9 +0,0 @@
-#!/bin/bash
-
-# Install the version of Node used in the platform tests on Jenkins
-# and then install the Node packages specified in the platform. These
-# installs will be cached and used as a baseline for installations
-# in future test runs.
-cd {{ jenkins_home }}/shallow-clone
-source scripts/jenkins-common.sh
-npm install
```
Comment (OmarEdraak) [inline playbooks/roles/jenkins_worker/templates/jenkins_nodeenv.j2]: Same here, I think we need this one.

File: playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml (hunk 1 above)
Comment (shadinaif) [reply playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml]: It is intended for caching. It should not brake anything
this is the original PR https://github.com/edx/configuration/pull/4779

Comment (shadinaif) [reply playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml]: If we keep it, I must resolve the failing mongo login here https://github.com/Edraak/edraak-platform/blob/032a62bd857551d5ad39368a994d46bf27deaf09/scripts/jenkins-common.sh#L14

Authentication is enabled in the DB on that point, but the line does not give an authentication. Unauthenticated login should be for replica DB only

what do you think @OmarEdraak ?

Comment (OmarEdraak) [reply playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml]: Makes sense!

**Comment by OmarEdraak** [comment]
> @OmarEdraak @devalih
> Please review this PR as Jenkins Worker setup did not succeed without these changes.
> 
> We have one problem as far as I can tell:
> 
> Edx team has upgraded the ansible version. Jenkins Worker is failing because of the new 2.3.1 version. This PR downgraded the version again, but this will brake something outside the Jenkins Worker setup.
> 
> We can stick with the new version; but that will require many updates on the workers yml files. Everything checking for a variable definition must be reviewed and many will need to be refactored. It is a bug in 2.3 version. Upgrading to 2.4 will solve this problem but it will create other problems

@shadinaif if it's about `when: ANALYTICS_API_DATABASES.default is defined` then you can convert it to:

```
 when:
   - ANALYTICS_API_DATABASES is defined
   - ANALYTICS_API_DATABASES.default is defined
```

It should work, does that help to avoid the Ansible version change?

**Comment by shadinaif** [comment]
@OmarEdraak  A developer on some thread said he tried this approach and it didn't work; so I didn't try it :man_facepalming: . But it is actually working as you expected :+1: 

I have kept the txt upgrade but restored ansible to 2.3.1. I will merge it now

**Comment by OmarEdraak** [comment]
@shadinaif if it's possible, try to not upgrade the packages unless absolutely necessary. This repository expects patches from upstream and upgrading this file will create git merge conflicts.

Reverting those changes is not a bad idea in my opinion.

**Comment by OmarEdraak** [comment]
@shadinaif you're an admin on this repository, you don't need to use your fork anymore.

**Pull request merged**